contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,792 @@
+        string calldata tokenIDsMapURI
+    ) external {
+        _updateSeries(seriesID, metadataURI, tokenIDsMapURI);
+    }
+
+    /**
+     * @dev Updates an existing series' metadata and token data
+     */
+    function batchUpdateSeries(
+        uint256[] calldata seriesIDs,
+        string[] calldata metadataURIs,
+        string[] calldata tokenIDsMapURIs
+    ) external {
+        _validateBatchParameters(seriesIDs.length, metadataURIs, tokenIDsMapURIs);
+        for (uint256 i = 0; i < seriesIDs.length; i++) {
+            _updateSeries(seriesIDs[i], metadataURIs[i], tokenIDsMapURIs[i]);
+        }
+    }
+
+    /**
+     * @dev Deletes series and cleans up related data
+     */
+    function deleteSeries(uint256 seriesID) external {
+        _deleteSeries(seriesID);
+    }
+
+    /**
+     * @dev Batch deletes series and cleans up related data
+     */
+    function batchDeleteSeries(uint256[] calldata seriesIDs) external {
+        for (uint256 i = 0; i < seriesIDs.length; i++) {
+            _deleteSeries(seriesIDs[i]);
+        }
+    }
+
+    // ============ Public/External Functions: Series Artist Management ============
+
+    /**
+     * @dev Allows an artist to remove themselves from a series
+     */
+    function removeSelfFromSeries(uint256 seriesID) external seriesExists(seriesID) onlyArtist(seriesID) {
+        uint256 artistID = addressToArtistID[msg.sender];
+
+        Series storage series = seriesDetails[seriesID];
+        uint256[] storage artistIDs = series.artistIDs;
+        
+        // Find and remove artistID from series' artist list
+        for (uint256 i = 0; i < artistIDs.length; i++) {
+            if (artistIDs[i] == artistID) {
+                artistIDs[i] = artistIDs[artistIDs.length - 1];
+                artistIDs.pop();
+                break;
+            }
+        }
+
+        // Remove series from artist's list
+        _removeSeriesFromArtist(artistID, seriesID);
+
+        emit SeriesUpdated(
+            seriesID, 
+            series.artistIDs, 
+            series.metadataURI, 
+            series.contractTokenDataURI
+        );
+    }
+
+    /**
+     * @dev Updates artist list for a series (owner only)
+     */
+    function ownerUpdateSeriesArtists(
+        uint256 seriesID,
+        address[] calldata artistAddrs
+    ) external seriesExists(seriesID) onlyOwner {
+        if (!ownerCanModifySeries(seriesID)) {
+            revert OwnerRightsRevokedForThisSeries(seriesID);
+        }
+        _validateArtistsNotRevoked(artistAddrs);
+
+        // Remove current artists
+        Series storage series = seriesDetails[seriesID];
+        for (uint256 i = 0; i < series.artistIDs.length; i++) {
+            uint256 artistID = series.artistIDs[i];
+            _removeSeriesFromArtist(artistID, seriesID);
+        }
+        delete series.artistIDs;
+
+        // Add new artists
+        _addArtistsToSeries(seriesID, artistAddrs);
+
+        emit SeriesUpdated(
```

Comment:
This event could be unclear. We just update the series artists so we don't need to fire the event with metadata URI or token data URI, I'd suggest to create new event if needed.